### PR TITLE
Test: Refactor some helpers to parallel

### PR DESCRIPTION
**Pull request by eloycoto**
Hi, 

Moved two functions to parallel to speed a little bit the tests. The functions are: 

- CiliumPolicyAction
- CiliumEndpointReady

Regards

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5261)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
For the track, 

Comparing the following builds: 

Master: https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/3735/execution/node/98/log/?consoleFull
Current build: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/5756/execution/node/93/log/

**Data:**

Before: 
```
13:07:36 • [SLOW TEST:171.301 seconds]
13:07:36 K8sPolicyTest
13:07:36 /home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:344
13:07:36   GuestBook Examples
13:07:36   /home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:344
13:07:36     checks policy example
13:07:36     /home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
```
After: 
```
15:03:12 • [SLOW TEST:110.683 seconds]
15:03:12 K8sPolicyTest
15:03:12 /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:344
15:03:12   GuestBook Examples
15:03:12   /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:344
15:03:12     checks policy example
15:03:12     /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
```

Before:
```
13:02:24 • [SLOW TEST:83.688 seconds]
13:02:24 K8sPolicyTest
13:02:24 /home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:344
13:02:24   Basic Test
13:02:24   /home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:344
13:02:24     Allows traffic with k8s default-allow ingress policy
13:02:24     /home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
```

After:
```
14:57:39 • [SLOW TEST:73.618 seconds]
14:57:39 K8sPolicyTest
14:57:39 /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:344
14:57:39   Basic Test
14:57:39   /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:344
14:57:39     Denies traffic with cnp default-deny egress policy
14:57:39     /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
```

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -830,6 +838,22 @@ func (kub *Kubectl) CiliumEndpointWaitReady() error {
 			}).Info("Waiting for cilium endpoints to be ready")
 
 			if invalid != 0 {
+				return
+			}
+			valid = true
+			return
+		}
+
+		for _, pod := range ciliumPods {
```
Comment (aanm) [inline test/helpers/kubectl.go]: wg.Add(len(ciliumPods))

File: test/helpers/kubectl.go [hunk 2]
```diff
@@ -984,37 +1008,54 @@ func (kub *Kubectl) CiliumPolicyAction(namespace, filepath string, action Resour
...
+			res := kub.CiliumExec(pod, fmt.Sprintf("cilium policy wait %d", revi))
+			if !res.WasSuccessful() {
+				kub.logger.Infof("CiliumPolicyAction: Policy wait failed on pod '%s': %s",
+					pod, res.GetStdOut())
+				return
+			}
+			status = true
+			return
 		}
+		for _, v := range pods {
```
Comment (aanm) [inline test/helpers/kubectl.go]: wg.Add(len(pods))

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by eloycoto** [comment]
test-me-please
EDIT: provision issue. 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/5772/testReport/junit/k8s-1/8/K8sTunnelTest_VXLan_Check_VXLAN_mode/

**Comment by eloycoto** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Pull request merged**